src/utilities/introspectionQuery.js

```diff
@@ -124,95 +124,117 @@ export type IntrospectionType =
   | IntrospectionEnumType
   | IntrospectionInputObjectType;
 
-export type IntrospectionScalarType = {
-  kind: 'SCALAR';
-  name: string;
-  description: ?string;
-};
-
-export type IntrospectionObjectType = {
-  kind: 'OBJECT';
-  name: string;
-  description: ?string;
-  fields: Array<IntrospectionField>;
-  interfaces: Array<IntrospectionNamedTypeRef>;
-};
-
-export type IntrospectionInterfaceType = {
-  kind: 'INTERFACE';
-  name: string;
-  description: ?string;
-  fields: Array<IntrospectionField>;
-  possibleTypes: Array<IntrospectionNamedTypeRef>;
-};
-
-export type IntrospectionUnionType = {
-  kind: 'UNION';
-  name: string;
-  description: ?string;
-  possibleTypes: Array<IntrospectionNamedTypeRef>;
-};
-
-export type IntrospectionEnumType = {
-  kind: 'ENUM';
-  name: string;
-  description: ?string;
-  enumValues: Array<IntrospectionEnumValue>;
-};
-
-export type IntrospectionInputObjectType = {
-  kind: 'INPUT_OBJECT';
-  name: string;
-  description: ?string;
-  inputFields: Array<IntrospectionInputValue>;
-};
+export type IntrospectionOutputType =
+  | IntrospectionScalarType
+  | IntrospectionObjectType
+  | IntrospectionInterfaceType
+  | IntrospectionUnionType
+  | IntrospectionEnumType;
+
+export type IntrospectionInputType =
+  | IntrospectionScalarType
+  | IntrospectionEnumType
+  | IntrospectionInputObjectType;
+
+export type IntrospectionScalarType = {|
+  +kind: 'SCALAR';
+  +name: string;
+  +description: ?string;
+|};
+
+export type IntrospectionObjectType = {|
+  +kind: 'OBJECT';
+  +name: string;
+  +description: ?string;
+  +fields: Array<IntrospectionField>;
+  +interfaces: Array<IntrospectionNamedTypeRef<IntrospectionInterfaceType>>;
+|};
+
+export type IntrospectionInterfaceType = {|
+  +kind: 'INTERFACE';
+  +name: string;
+  +description: ?string;
+  +fields: Array<IntrospectionField>;
+  +possibleTypes: Array<IntrospectionNamedTypeRef<IntrospectionObjectType>>;
+|};
+
+export type IntrospectionUnionType = {|
+  +kind: 'UNION';
+  +name: string;
+  +description: ?string;
+  +possibleTypes: Array<IntrospectionNamedTypeRef<IntrospectionObjectType>>;
+|};
+
+export type IntrospectionEnumType = {|
+  +kind: 'ENUM';
+  +name: string;
+  +description: ?string;
+  +enumValues: Array<IntrospectionEnumValue>;
+|};
+
+export type IntrospectionInputObjectType = {|
+  +kind: 'INPUT_OBJECT';
+  +name: string;
+  +description: ?string;
+  +inputFields: Array<IntrospectionInputValue>;
+|};
+
+export type IntrospectionListTypeRef<T> = {|
+  +kind: 'LIST',
+  +ofType?: T,
+|};
+
+export type IntrospectionNonNullTypeRef<T> = {|
```

Comment:
`<T: IntrospectionType = IntrospectionType>`

same above. That means supplying T must be a subtype of IntrospectionType, but you can also omit <T>, which just implies IntrospectionType - this is nice because it means this won't be a breaking change for anyone already using these types

Reply:
Ahhh nice